packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2440,6 +2441,12 @@ export class SnapController extends BaseController<
       'ExecutionService:outboundResponse',
       this._onOutboundResponse,
     );
+
+    this.messagingSystem.clearEventSubscriptions(
```

Comment:
Should we do this for every event? Why is this only popping up now?

Reply:
This was unrelated to the actual issue with the tests getting stuck. I just noticed that we didn't properly unsubscribe these events.

Reply:
Let's either do it in another PR or do it for every event here! 😅

Reply:
Well, I only did it for these events because we subscribe to them in the Snap controller. I don't think we need to do it for every event.